### ISSUE TITLE
Unhandled error causes crash - Cache strategy sorted

### ISSUE THREAD

**Issue by jpzk** [graphite-project/carbon#828]
## Scenario

I'm running this https://github.com/graphite-project/docker-graphite-statsd docker image (this one is running Carbon 1.1.4 (latest version, at the time of writing). And sometimes I get this error, which causes carbon to crash. Could the reason be, that I take a backup of the whisper folder while running (maybe corrupting the data?)

EDIT: looks like a cache.py related issue?

## Error log

```
01/12/2018 18:49:20 :: [console] Unhandled Error
Traceback (most recent call last):
 File “/opt/graphite/lib/python3.6/site-packages/twisted/python/threadpool.py”, line 250, in inContext
   result = inContext.theWork()
 File “/opt/graphite/lib/python3.6/site-packages/twisted/python/threadpool.py”, line 266, in <lambda>
   inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
 File “/opt/graphite/lib/python3.6/site-packages/twisted/python/context.py”, line 122, in callWithContext
   return self.currentContext().callWithContext(ctx, func, *args, **kw)
 File “/opt/graphite/lib/python3.6/site-packages/twisted/python/context.py”, line 85, in callWithContext
   return func(*args,**kw)
--- <exception caught here> ---
 File “/opt/graphite/lib/carbon/writer.py”, line 189, in writeForever
   writeCachedDataPoints()
 File “/opt/graphite/lib/carbon/writer.py”, line 98, in writeCachedDataPoints
   (metric, datapoints) = cache.drain_metric()
 File “/opt/graphite/lib/carbon/cache.py”, line 187, in drain_metric
   metric = self.strategy.choose_item()
 File “/opt/graphite/lib/carbon/cache.py”, line 116, in choose_item
   return next(self.queue)
builtins.StopIteration:
```

## Additional notes

This one is *not* related to https://github.com/graphite-project/carbon/issues/766

**Comment by deniszh** [comment]
Duplicate of https://github.com/graphite-project/carbon/issues/815
Fixed in https://github.com/graphite-project/carbon/pull/825
Will be released in 1.1.5 soon (preparing release already, this week probably).
Current workaround: do not use "sorted" and python 3 together.

**Comment by jpzk** [comment]
Nice, thanks for the fast response! Awesome support.

**Comment by deniszh** [comment]
Fix released in 1.1.5
